### PR TITLE
Add Command Line Reference section to docs

### PR DESCRIPTION
**Pull request by atugushev**
Closes https://github.com/jazzband/pip-tools/issues/1633
Addresses https://github.com/jazzband/pip-tools/issues/1845#issuecomment-1544012577.
Pre-view available on: https://pip-tools--1934.org.readthedocs.build/en/1934/

cc: @chrysle 

##### Contributor checklist

- [ ] Provided the tests for the changes.
- [x] Assure PR title is short, clear, and good to be included in the user-oriented changelog

##### Maintainer checklist

- [x] Assure one of these labels is present: `backwards incompatible`, `feature`, `enhancement`, `deprecation`, `bug`, `dependency`, `docs` or `skip-changelog` as they determine changelog listing.
- [ ] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

File: docs/cli-reference/index.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# CLI Reference
```
Comment (chrysle) [inline docs/cli-reference/index.md:1]:
```suggestion
# Command Line Reference
```
I don't think we need to use the short form here?

**Review by chrysle — APPROVED** [review]
LGTM, thanks for doing my work! ;-)

File: docs/cli-reference/index.md (hunk 1 above)
Comment (atugushev) [reply docs/cli-reference/index.md]: Valid point. Thanks!

**Comment by atugushev** [comment]
Thanks @chrysle!

**Pull request merged**